libsql-server/src/hrana/ws/session.rs

```diff
@@ -65,60 +88,22 @@ pub enum ResponseError {
     Batch(batch::BatchError),
 }
 
-pub(super) async fn handle_initial_hello(
+pub(super) async fn handle_hello(
     server: &Server,
-    version: Version,
     jwt: Option<String>,
     namespace: NamespaceName,
-) -> Result<Session> {
-    // todo dupe #auth
+) -> Result<Authenticated> {
     let namespace_jwt_key = server
         .namespaces
         .with(namespace.clone(), |ns| ns.jwt_key())
         .await??;
 
-    let auth = namespace_jwt_key
-        .map(Jwt::new)
-        .map(Auth::new)
-        .unwrap_or(server.user_auth_strategy.clone())
-        .authenticate(Ok(UserAuthContext::bearer_opt(jwt)))
-        .map_err(|err| anyhow!(ResponseError::Auth { source: err }))?;
-
-    Ok(Session {
-        auth,
-        version,
-        streams: HashMap::new(),
-        sqls: HashMap::new(),
-        cursors: HashMap::new(),
-    })
-}
-
-pub(super) async fn handle_repeated_hello(
-    server: &Server,
-    session: &mut Session,
-    jwt: Option<String>,
-    namespace: NamespaceName,
-) -> Result<()> {
-    if session.version < Version::Hrana2 {
-        bail!(ProtocolError::NotSupported {
-            what: "Repeated hello message",
-            min_version: Version::Hrana2,
-        })
-    }
-    // todo dupe #auth
-    let namespace_jwt_key = server
-        .namespaces
-        .with(namespace.clone(), |ns| ns.jwt_key())
-        .await??;
-
-    session.auth = namespace_jwt_key
+    namespace_jwt_key
```

Comment:
nice 👍

Reply:
This is already in main @LucioFranco -> https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/hrana/ws/session.rs#L101